src/main/java/cubicchunks/asm/mixin/core/client/MixinRenderGlobal_BiggerRenderChunk_Common.java

```diff
@@ -0,0 +1,230 @@
+    @Shadow private ViewFrustum viewFrustum;
+    @Shadow @Final private Minecraft mc;
+    @Shadow private WorldClient world;
+    @Shadow private int renderEntitiesStartupCounter;
+    @Shadow @Final private RenderManager renderManager;
+    @Shadow private int countEntitiesTotal;
+    @Shadow private int countEntitiesRendered;
+    @Shadow private int countEntitiesHidden;
+    
+    @Shadow
+    private boolean isOutlineActive(Entity entityIn, Entity viewer, ICamera camera) {
+        throw new AbstractMethodError();
+    };
+    
+    @Shadow
+    public abstract void loadRenderers();
+    
+    private int renderChunkSizeBit = -1;
+    
+    ClassInheritanceMultiMap<Entity> mutableEntityMapWrapper = new ClassInheritanceMultiMap<Entity>(Entity.class);
+    @SuppressWarnings("unchecked") ClassInheritanceMultiMap<Entity>[] dummyEntityStorageArray = new ClassInheritanceMultiMap[1];
+
+    @Redirect(method = "renderEntities", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/renderer/chunk/RenderChunk;getPosition()Lnet/minecraft/util/math/BlockPos;"))
+    public BlockPos onGetPosition(RenderChunk renderChunk) {
+        int renderChunkCubeSize = RenderVariables.getRenderChunkSize() / Cube.SIZE;
+        mutableEntityMapWrapper.values.clear();
+        BlockPos rcpos = renderChunk.getPosition();
+        int cx0 = Coords.blockToCube(rcpos.getX());
+        int cy0 = Coords.blockToCube(rcpos.getY());
+        int cz0 = Coords.blockToCube(rcpos.getZ());
+        ICubicWorldClient cworld = (ICubicWorldClient) world;
+        if (cworld.isCubicWorld()) {
+            CubeProviderClient cubeProvider = cworld.getCubeCache();
+            for (int cx = cx0; cx < cx0 + renderChunkCubeSize; cx++)
+                for (int cy = cy0; cy < cy0 + renderChunkCubeSize; cy++)
+                    for (int cz = cz0; cz < cz0 + renderChunkCubeSize; cz++) {
+                        mutableEntityMapWrapper.values.addAll(cubeProvider.getCube(cx, cy, cz).getEntityContainer().getEntitySet());
+                    }
+        } else {
+            for (int cx = cx0; cx < cx0 + renderChunkCubeSize; cx++)
+                for (int cz = cz0; cz < cz0 + renderChunkCubeSize; cz++) {
+                    ClassInheritanceMultiMap<Entity>[] entityLists = world.getChunkFromChunkCoords(cx, cz).getEntityLists();
+                    for (int cy = cy0; cy < cy0 + renderChunkCubeSize && cy < 16 && cy >= 0; cy++) {
+                        mutableEntityMapWrapper.values.addAll(entityLists[cy].values);
+                    }
+                }
+        }
+        return BlockPos.ORIGIN;
+    }
+
+    @Redirect(method = "renderEntities", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/chunk/Chunk;getEntityLists()[Lnet/minecraft/util/ClassInheritanceMultiMap;"))
+    public ClassInheritanceMultiMap<Entity>[] onGettingClassInheritanceMultiMap(Chunk chunk) {
+        dummyEntityStorageArray[0] = mutableEntityMapWrapper;
+        return dummyEntityStorageArray;
```

Comment:
If the value of `mutableEntityMapWrapper` never changes, why can't you just `return dummyEntityStorageArray` and set the array value once?

Reply:
I could. I just need a one more inject in class constructor.

Reply:
Can't it be done in field initializer?

Reply:
No. Mixin throw error.